src/MLJClusteringInterface.jl

```diff
@@ -193,6 +193,155 @@ metadata_model(
     descr = KMedoidsDescription,
     path = "$(PKG).KMedoids"
 )
+"""
+$(MMI.doc_header(KMeans))
 
-end # module
 
+`KMeans`: The K-Means algorithm finds K centroids corresponding to K clusters in
+the data. The clusters are assumed to be elliptical, should be used with a euclidean distance metric
+
+# Training data
+
+In MLJ or MLJBase, bind an instance `model` to data with
+
+    mach = machine(model, X, y)
+
+Where
+
+- `X`: is any table of input features (eg, a `DataFrame`) whose columns
+  are of scitype `Continuous`; check the scitype with `schema(X)`
+
+- `y`: is the target, which can be any `AbstractVector` whose element
+  scitype is `Count`; check the scitype with `schema(y)`
+
+Train the machine using `fit!(mach, rows=...)`.
+
+# Hyper-parameters
+
+- `k=3`: The number of centroids to use in clustering.
+- `metric::SemiMetric=SqEuclidean`: The metric used to calculate the clustering distance
+  matrix
+
+# Operations
+
+- `predict(mach, Xnew)`: return predictions of the target given new
+   features `Xnew` having the same Scitype as `X` above.
```

Comment:
Let's clarify this a bit, as this is unsupervised classification. A suggestion:

```suggestion
- `predict(mach, Xnew)`: return learned cluster labels for  a new
   table of inputs `Xnew` having the same scitype as `X` above.
```